LayerClient/LayerClient.py

```diff
@@ -0,0 +1,519 @@
+        - `message_parts`: An array of `LayerClient.MessagePart` objects
+        - `notification`: Optional `PushNotification` instance.
+        """
+        request_data = {
+            'sender': {
+                'name': sender.name,
+            },
+            'parts': [
+                part.as_dict() for part in message_parts
+            ],
+            'recipients': recipients,
+        }
+        if notification:
+            request_data['notification'] = notification.as_dict()
+
+        return Announcement.from_dict(
+            self._raw_request(
+                METHOD_POST,
+                self._get_layer_uri(LAYER_URI_ANNOUNCEMENTS),
+                request_data,
+            )
+        )
+
+
+class Announcement(BaseLayerResponse):
+    """
+    Contains the data returned from the API when sending an Announcement
+    """
+
+    def __init__(self, id, url, sent_at, recipients, sender, parts):
+        self.id = id
+        self.url = url
+        self.sent_at = sent_at
+        self.recipients = recipients
+        self.sender = sender
+        self.parts = parts
+
+    @staticmethod
+    def from_dict(dict_data):
+        return Announcement(
+            dict_data.get('id'),
+            dict_data.get('url'),
+            Announcement.parse_date(dict_data.get('sent_at')),
+            dict_data.get('recipients'),
+            Sender.from_dict(dict_data.get('sender')),
+            [
+                MessagePart.from_dict(part) for part in dict_data.get('parts')
+            ],
+        )
+
+    def __repr__(self):
+        return '<LayerClient.Announcement "{text}">'.format(
+            text=self.uuid()
+        )
+
+
+class Message(BaseLayerResponse):
+    """
+    The response returned by the API when a message is sent.
+    """
+
+    def __init__(self, id, url, sent_at=None, sender=None,
+                 conversation=None, parts=None, recipient_status=None):
+        self.id = id
+        self.url = url
+        self.sent_at = sent_at
+        self.sender = sender
+        self.conversation = conversation
+        self.parts = parts
+        self.recipient_status = recipient_status
+
+    @staticmethod
+    def from_dict(dict_data):
+        return Message(
+            dict_data.get('id'),
+            dict_data.get('url'),
+            Message.parse_date(dict_data.get('sent_at')),
+            Sender.from_dict(dict_data.get('sender')),
+            Conversation.from_dict(dict_data.get('conversation')),
+            [
+                MessagePart.from_dict(part) for part in dict_data.get('parts')
+            ],
+            dict_data.get('recipient_status'),
+        )
+
+    def __repr__(self):
+        return '<LayerClient.Message "{uuid}">'.format(
+            uuid=self.uuid()
+        )
+
+
+class Sender:
+    """
+    Used for sending messages.
+    Id and Name may both be set, but the send_message API will prefer
+    one over the other.
+    """
+
+    def __init__(self, id=None, name=None):
```

Comment:
Does there need to be at least one? Seems like you might want to prevent the case where they are both none